cpp/src/replace/replace.cu

```diff
@@ -13,26 +16,551 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 #include <thrust/device_ptr.h>
 #include <thrust/find.h>
 #include <thrust/execution_policy.h>
-
-#include <cudf/cudf.h>
+#include <cub/cub.cuh>
+#include <cudf/copying.hpp>
+#include <cudf/replace.hpp>
+#include <cudf/detail/replace.hpp>
 #include <rmm/rmm.h>
 #include <cudf/types.hpp>
-#include <cudf/utilities/error.hpp>
-#include <cudf/utilities/legacy/type_dispatcher.hpp>
-#include <cudf/column/column_device_view.cuh>
-#include <cudf/column/column_factories.hpp>
+#include <cudf/scalar/scalar.hpp>
 #include <cudf/utilities/type_dispatcher.hpp>
 #include <utilities/cudf_utils.h>
 #include <utilities/cuda_utils.hpp>
 #include <utilities/column_utils.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/utilities/bit.hpp>
+#include <cudf/column/column_device_view.cuh>
+
+namespace{ //anonymous
+
+static constexpr int warp_size = 32;
+static constexpr int BLOCK_SIZE = 256;
+
+// returns the block_sum using the given shared array of warp sums.
+template <typename T>
+__device__ T sum_warps(T* warp_smem)
```

Comment:
This looks like it should be replaced by https://github.com/rapidsai/cudf/blob/branch-0.11/cpp/include/cudf/detail/utilities/cuda.cuh#L82

Reply:
These functions don't do quite the same thing. Could you provide more guidance on how to use?

Reply:
Done.

Reply:
So `sum_warps` is not called anymore? If not, please remove it.

Reply:
Removed it.